xpk.py

```diff
@@ -1666,6 +1741,182 @@ def add_roles_to_service_account(args) -> int:
   return 0
 
 
+def create_autoprovisioning_config(args, system: SystemCharacteristics) -> tuple[AutoprovisioningConfig | None, int]:
+  """Create autoprovisioning config based on template file and user args
+
+  Args:
+    args: user provided arguments for running the command.
+    system: system characteristics.
+
+  Returns:
+    tuple[AutoprovisioningConfig, int]
+    AutoprovisioningConfig: config used to enable autoprovisioning
+    int: return code
+  """
+
+  # CPU Limits and Memory Limits are for user jobs only. The default node pool
+  # is not controlled by NAP.
+  cpu_limits = """
+  minimum: 1
+  maximum: 10000
+  """
+  memory_limits = """
+  minimum: 1
+  maximum: 10000
+  """
+
+  # By default, the maximum chips is set to be the current number of resources used
+  # in the cluster. The minimum is set to zero.
+  if system.accelerator_type == AcceleratorType['GPU']:
+    num_chips_in_cluster = system.vms_per_slice * system.chips_per_vm * args.num_nodes
```

Comment:
Why is this calculation different from line 2868?

Reply:
This should be the same, created a new helper function to reduce the dup code: determine_total_chips_from_args